src/com/ra4king/circuitsim/simulator/components/memory/RAM.java

```diff
@@ -90,6 +99,8 @@ public void valueChanged(CircuitState state, WireValue value, int portIndex) {
 		boolean load = state.getLastReceived(getPort(PORT_LOAD)).getBit(0) != State.ZERO;
 		boolean clear = state.getLastReceived(getPort(PORT_CLEAR)).getBit(0) == State.ONE;
 		
+		boolean store = state.getLastReceived(getPort(PORT_STORE)).getBit(0) != State.ZERO;
```

Comment:
Once you make the constructor change, this will throw an exception for not separate load/store. Get the value inside the if statement.

Reply:
I handled this by boolean shortcircuit with &&ing isSeparateLoadStore with this. If you'd rather not lmk